examples/sampling.py

```diff
@@ -0,0 +1,247 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+"""
+=========================
+How to sample video clips
+=========================
+
+In this example, we'll learn how to sample video :term:`clips` from a video. A
+clip generally denotes a sequence or batch of frames, and is typically passed as
+input to video models.
+"""
+
+# %%
+# First, a bit of boilerplate: we'll download a video from the web, and define a
+# plotting utility. You can ignore that part and jump right below to
+# :ref:`sampling_tuto_start`.
+
+from typing import Optional
+import torch
+import requests
+
+
+# Video source: https://www.pexels.com/video/dog-eating-854132/
+# License: CC0. Author: Coverr.
+url = "https://videos.pexels.com/video-files/854132/854132-sd_640_360_25fps.mp4"
+response = requests.get(url, headers={"User-Agent": ""})
+if response.status_code != 200:
+    raise RuntimeError(f"Failed to download video. {response.status_code = }.")
+
+raw_video_bytes = response.content
+
+
+def plot(frames: torch.Tensor, title : Optional[str] = None):
+    try:
+        from torchvision.utils import make_grid
+        from torchvision.transforms.v2.functional import to_pil_image
+        import matplotlib.pyplot as plt
+    except ImportError:
+        print("Cannot plot, please run `pip install torchvision matplotlib`")
+        return
+
+    plt.rcParams["savefig.bbox"] = 'tight'
+    fig, ax = plt.subplots()
+    ax.imshow(to_pil_image(make_grid(frames)))
+    ax.set(xticklabels=[], yticklabels=[], xticks=[], yticks=[])
+    if title is not None:
+        ax.set_title(title)
+    plt.tight_layout()
+
+
+# %%
+# .. _sampling_tuto_start:
+#
+# Creating a decoder
+# ------------------
+#
+# Sampling clips from a video always starts by creating a
+# :class:`~torchcodec.decoders.VideoDecoder` object. If you're not already
+# familiar with :class:`~torchcodec.decoders.VideoDecoder`, take a quick look
+# at: :ref:`sphx_glr_generated_examples_basic_example.py`.
+from torchcodec.decoders import VideoDecoder
+
+# You can also pass a path to a local file!
+decoder = VideoDecoder(raw_video_bytes)
+
+# %%
+# Sampling basics
+# ---------------
+#
+# We can now use our decoder to sample clips. Let's first look at a simple
+# example: all other samplers follow similar APIs and principles. We'll use
+# :func:`~torchcodec.samplers.clips_at_random_indices` to sample clips that
+# start at random indices.
+
+from torchcodec.samplers import clips_at_random_indices
+
+torch.manual_seed(0)  # The samplers RNG is controlled by pytorch's RNG
```

Comment:
Relevant to all of our recent discussion, I think we should have a more substantial comment here. Something like:
```
# We're setting the seed to a fixed value here because we want to get reproducible results
# everytime we run the code in this tutorial. In general, you should do the same if you also 
# want reproducible results, but it is not an actual requirement for using our API.
```